### DISCUSSION TITLE
API token is invalid only when I try to update a page properties.

### DISCUSSION THREAD

**Discussion by kwokrico** [ramnes/notion-sdk-py#239] [Q&A] [answered]
I am facing a strange problem here. I have no problem using notion sdk to create, query and delete pages, but when it comes to updating a page properties, it always results in a 401 Error: API token is invalid. I'm really confused whether I did something wrong with my codes, token setup, wrong formatting of the properties to be updated or it's just a bug. 

The images below are my codes and the logs. Much appreciated if anyone can help me solve this problem here. Thanks in advance. 

![image](https://github.com/user-attachments/assets/4a70b9a1-53e7-4315-8432-79f37818340a)

![image](https://github.com/user-attachments/assets/00815f6e-87b3-426f-8664-81433e69d529)
![image](https://github.com/user-attachments/assets/eb5c5466-7295-4d5f-a26a-6cb40f9b1705)
![image](https://github.com/user-attachments/assets/c43228c9-2db5-4a1d-804d-b88d1ea3d53e)

**Comment by ramnes** [comment, marked as answer]
I think you have two problems:
* Your update parameters are invalid (there's at least `properties={"properties": ...}}` that seems wrong at first sight) – please refer to [the Notion API reference](https://developers.notion.com/reference/patch-page);
* HTTPX seems out of date, because `httpx.HTTPStatusError` should exist since version 0.14.0 (August 7th, 2020).

What version do you use?

**Reply by kwokrico** [reply]
I have solved the problem, after looking into your answer. Thanks. There are a few helpful links below in case anyone facing a similar problem in the future. 

For 1, I found the [Page properties](https://developers.notion.com/reference/page-property-values#url) and [Property values](https://developers.notion.com/reference/property-value-object) in the [the Notion API reference](https://developers.notion.com/reference/patch-page). 
Noted that in page properties, there is a properties={"properties": ...}} in its Example properties body param for a POST or PATCH page in the page properties page (but it is not needed in my final code) (Maybe I misunderstood something there)
![image](https://github.com/user-attachments/assets/d8a97ff0-91ef-45b3-8357-118b7d9889a0)

For 2, updating the library solves my problem.

Anyway. Thank you for the help.
